src/features/content-explorer/content-explorer/ContentExplorer.js

```diff
@@ -383,14 +415,44 @@ class ContentExplorer extends Component {
     };
 
     handleSelectAllClick = async () => {
-        // check if the items list is still loading
-        const { items } = this.props;
-        if (items && items[0] && items[0].isLoading) {
+        const { items, onSelectedItems } = this.props;
+        if (isLoadingItems(items)) {
             return;
         }
         const { isSelectAllChecked } = this.state;
         const newSelectedItems = isSelectAllChecked ? this.unselectAll() : this.selectAll();
         this.setState({ selectedItems: newSelectedItems, isSelectAllChecked: !isSelectAllChecked });
+        if (onSelectedItems) {
+            onSelectedItems(newSelectedItems);
+        }
+    };
+
+    numOfSelectedItems = () => {
+        const { selectedItems } = this.state;
+        return Object.keys(selectedItems).length;
+    };
+
+    handleIncludeSubfoldersToggle = () => {
+        const { isIncludeSubfoldersEnabled } = this.state;
+        const { onIncludeSubfoldersToggle } = this.props;
+        this.setState({ isIncludeSubfoldersEnabled: !isIncludeSubfoldersEnabled });
+        onIncludeSubfoldersToggle();
+    };
+
+    includeSubfolderToggleDisabled = () => {
+        const { isIncludeSubfoldersEnabled, selectedItems } = this.state;
+        const { items } = this.props;
+        const selectedItemIsFolder = this.numOfSelectedItems() === 1 && isFolderPresent(Object.values(selectedItems));
+        const selectedItemIsNotFolder =
+            this.numOfSelectedItems() === 1 && !isFolderPresent(Object.values(selectedItems));
+        return (
+            isLoadingItems(items) ||
+            (!isIncludeSubfoldersEnabled &&
+                !(selectedItemIsFolder && !isFolderPresent(items)) &&
+                ((!areAllItemsActionDisabled(items) && !isFolderPresent(items)) ||
+                    selectedItemIsNotFolder ||
+                    this.numOfSelectedItems() > 1))
+        );
```

Comment:
this logic is hard to follow and as well as could be different by use case. it looks we can determine if the toggle should be disabled in the parent app because the wrapping component should know the `selectedItems` and can track the state of the toggle using `onIncludeSubfoldersToggle`.

we might be able to get away with passing only one prop all the way down called `includeSubfoldersProps` (similar to `searchInputProps`).

the wrapping component should keep it's own state of if the toggle is enabled and the selected items